datafusion/core/src/physical_optimizer/sort_enforcement.rs

```diff
@@ -146,19 +151,11 @@ impl PlanWithCorrespondingSort {
                     return None;
                 }
                 let is_spm = is_sort_preserving_merge(plan);
-                let is_union = plan.as_any().is::<UnionExec>();
```

Comment:
Since we now have a pushdown subroutine. We no longer need special handling for union here.

Reply:
I like the idea of separting the "enforce sort requirements" from the "optimize sort requirements" which I think this PR improves